[addons] change binary callbacks

AlwinEsch opened the pull request:
## Description
----------------------
1. Add binary addon callback to get used skin name id

This add a callback who also present on python and becomes needed
on e.g. Web browser addon to know used skin name.

The change is more general and included to a new request.

----------------------
2. allow binary addon to use label id with every id

This change the id usage of strings on new binary addons to override
a kodi string id with a own for him itself.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons] allow binary addon to use label id with every id

This change the id usage of strings on new binary addons to override
a kodi string id with a own for him itself.

tamland commented on xbmc/addons/interfaces/General.cpp:
```diff
@@ -413,6 +412,18 @@ int Interface_General::get_global_idle_time(void* kodiBase)
+    return nullptr;
+  }
+
+  return strdup(CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOOKANDFEEL_SKIN).c_str());
```
The setting stores the skin id, not the name. I'm guessing you actually want is the id; if so the methods should say 'id'

AlwinEsch replied:
Is changed to use id instead of name

AlwinEsch committed:
[addons] add binary addon callback to get used skin name

This add a callback who also present on python and becomes needed
on e.g. Web browser addon to know used skin name.

The change is more general and included to a new request.

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please

AlwinEsch commented:
jenkins build this please

ksooo commented:
You know that I always get puzzled about this version number stuff, but imo this change breaks all old style binary add-ons, like pvr add-ons. @AlwinEsch

@MilhouseVH not sure whether I'm actually right, be in case I am, you are informed...

ksooo commented:
Okay, you did not modify the function tables, right? Thus, all should be fine, but I would be happy if you could confirm this.

AlwinEsch commented:
PVR still use the old and not affected by them. Also is with a add of callback function (if added to the end) still backward compatible.

ksooo commented:
Sure, thanks.
